components/general/AdapterComponent.js

```diff
@@ -8,11 +8,21 @@ var PythonControlledTextField = PythonControlledCapability.createPythonControlle
  * The slider bar can have a set minimum and maximum, and the value can be
  * obtained through the value parameter fired on an onChange event.
  */
+
 export default class AdapterComponent extends Component {
 
     constructor(props) {
         super(props);
-        this.state = {};
+        /** 
+         * the state is constructed dynamically from the id props of each children
+         * in this way we are declaring a controlled component that can handle his own
+         * state when this is modified by a new input or action of the user
+         **/
+        this.stateBuilder = {};
+        this.props.children.map( (child, index) => {
```

Comment:
not sure if it really matter but maybe we can use **forEach** instead of **map** because we don't want to return anything.